hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecord.java

```diff
@@ -427,6 +428,24 @@ public HoodieRecord rewriteRecordWithNewSchema(Schema recordSchema, Properties p
 
   public abstract Option<Map<String, String>> getMetadata();
 
+  private void initMetadata() {
+    this.metaData = Option.of(new HashMap<>());
```

Comment:
`this.metaData` is not included in the serialization and deserialization so the information can be lost.